src/App.tsx

```diff
@@ -21,75 +22,144 @@ enum SortType {
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
   // eslint-disable-next-line no-console
   console.log(sortType, isReversed);
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+interface State {
+  isReversed: boolean,
+  sortType: SortType,
+}
+
+export class App extends React.Component<{}, State> {
+  state: State = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  hendleAlphabeticallySort = () => {
+    this.setState({
+      sortType: SortType.ALPHABET,
+    });
+  };
+
+  hendleLengthSort = () => {
+    this.setState({
+      sortType: SortType.LENGTH,
+    });
+  };
+
+  handleReset = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+    });
+  };
+
+  handleReverse = () => {
+    this.setState(state => ({
+      isReversed: !state.isReversed,
+    }));
+  };
+
+  render() {
+    const { isReversed, sortType } = this.state;
+    const {
+      hendleAlphabeticallySort,
+      hendleLengthSort,
+      handleReverse,
+      handleReset,
+    } = this;
+
+    const goods = getReorderedGoods(goodsFromServer, this.state);
+    const isSorted = sortType !== SortType.NONE;
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={cn('button', 'is-info', {
+              'is-light': sortType !== SortType.ALPHABET,
+            })}
+            onClick={hendleAlphabeticallySort}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={cn('button', 'is-success', {
+              'is-light': sortType !== SortType.LENGTH,
+            })}
+            onClick={hendleLengthSort}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={cn('button', 'is-warning', {
+              'is-light': !isReversed,
+            })}
+            onClick={handleReverse}
+          >
+            Reverse
+          </button>
+
+          {(isSorted || isReversed)
+            && (
```

Comment:
move to the separate variable
```suggestion
          {  ... && (
```